core/src/display_object/edit_text.rs

```diff
@@ -1657,27 +1657,33 @@ impl<'gc> TDisplayObject<'gc> for EditText<'gc> {
     }
 
     fn width(&self) -> f64 {
-        self.0.read().bounds.width().to_pixels()
+        let edit_text = self.0.read();
+        let text_width = edit_text.bounds.width();
+        text_width.to_pixels() * (edit_text.base.transform.matrix.a as f64)
     }
 
     fn set_width(&self, gc_context: MutationContext<'gc, '_>, value: f64) {
         let mut write = self.0.write(gc_context);
+        let scaled_value = value / (write.base.transform.matrix.a as f64);
 
-        write.bounds.set_width(Twips::from_pixels(value));
+        write.bounds.set_width(Twips::from_pixels(scaled_value));
```

Comment:
I think this should change the bounds directly without affecting the matrix; the matrix will transform the bounds additionally during rendering. For example:
```
import flash.geom.Matrix;
trace(txt.width);  // 100
var matrix:Matrix = new Matrix();
matrix.a = 1.5;
txt.transform.matrix = matrix;
trace(txt.transform.matrix); // a = 1.5
trace(txt.width);            // 150
txt.width = 50;	
trace(txt.width);			 // 75	
trace(txt.transform.matrix); // still a = 1.5
// internally the bounds will have width = 50, multiplied 1.5 when rendering
```